cpp/oneapi/dal/algo/knn/backend/gpu/train_kernel_brute_force_dpc.cpp

```diff
@@ -0,0 +1,110 @@
+* limitations under the License.
+*******************************************************************************/
+
+#define DAAL_SYCL_INTERFACE
+
+#include "algorithms/engines/mcg59/mcg59.h"
+#include "data_management/data/numeric_table.h"
+#include "oneapi/dal/backend/interop/table_conversion.hpp"
+#include "src/algorithms/k_nearest_neighbors/oneapi/bf_knn_classification_model_ucapi_impl.h"
+#include "src/algorithms/k_nearest_neighbors/oneapi/bf_knn_classification_train_kernel_ucapi.h"
+
+#include "oneapi/dal/algo/knn/backend/gpu/train_kernel.hpp"
+#include "oneapi/dal/algo/knn/backend/model_interop.hpp"
+#include "oneapi/dal/algo/knn/detail/model_impl.hpp"
+#include "oneapi/dal/backend/dispatcher_dpc.hpp"
+#include "oneapi/dal/backend/interop/common_dpc.hpp"
+#include "oneapi/dal/backend/interop/error_converter.hpp"
+
+namespace oneapi::dal::knn::backend {
+
+using dal::backend::context_gpu;
+using namespace daal::data_management;
+
+namespace daal_knn = daal::algorithms::bf_knn_classification;
+namespace interop  = dal::backend::interop;
+
+template <typename Float>
+using daal_knn_brute_force_kernel_t =
+    daal_knn::training::internal::KNNClassificationTrainKernelUCAPI<Float>;
+using daal_interop_model_t = detail::model_impl::interop_model;
+
+template <typename Float>
+static train_result call_daal_kernel(const context_gpu& ctx,
+                                     const descriptor_base& desc,
+                                     const table& data,
+                                     const table& labels) {
+    auto queue = ctx.get_queue();
+    interop::execution_context_guard guard(queue);
+
+    const std::int64_t row_count    = data.get_row_count();
+    const std::int64_t column_count = data.get_column_count();
+
+    auto arr_data   = row_accessor<const Float>{ data }.pull();
+    auto arr_labels = row_accessor<const Float>{ labels }.pull();
+
+    const auto daal_data =
+        interop::convert_to_daal_sycl_homogen_table(queue, arr_data, row_count, column_count);
+    const auto daal_labels =
+        interop::convert_to_daal_sycl_homogen_table(queue, arr_labels, row_count, 1);
+
+    daal_knn::Parameter daal_parameter(
+        desc.get_class_count(),
+        desc.get_neighbor_count(),
+        desc.get_data_use_in_model() ? daal_knn::doUse : daal_knn::doNotUse);
+
+    daal::algorithms::classifier::ModelPtr model_ptr(new daal_knn::Model(column_count));
+    if (!model_ptr) {
+        throw bad_alloc();
+    }
+
+    auto knn_model = static_cast<daal_knn::Model*>(model_ptr.get());
+
+    knn_model->impl()->setData<Float>(daal_data, desc.get_data_use_in_model());
+    knn_model->impl()->setLabels<Float>(daal_labels, desc.get_data_use_in_model());
+
+    interop::status_to_exception(daal_knn_brute_force_kernel_t<Float>().compute(
+        daal_data.get(),
+        daal_labels.get(),
+        knn_model,
+        daal_parameter,
+        *(daal::algorithms::engines::mcg59::Batch<>::create())));
```

Comment:
the generator is already created in the parameter.
```c++
    Parameter(size_t nClasses = 2, size_t nNeighbors = 1, int randomSeed = 777, DataUseInModel dataUse = doNotUse)
        : daal::algorithms::classifier::Parameter(nClasses),
          k(nNeighbors),
          seed(randomSeed),
          dataUseInModel(dataUse),
          engine(engines::mcg59::Batch<>::create())
    {}
```
Use him:
```suggestion
        *par->engine));
```